### PR TITLE
[04/08] replace read-only git query paths with gitoxide

### PR DESCRIPTION
**Pull request by mjc**
Stacked on `mjc/gitoxide-spike-03-graph-walker`.

Why this PR exists:

- Earlier stack layers moved graph walking and commit identity toward gitoxide.
- This PR applies that same boundary shift to read-only query surfaces used by status panes, file search, remotes, submodules, worktrees, branch visibility, and related UI state.
- The goal is to reduce read-only query dependence on libgit2 object/status plumbing while keeping mutating Git actions for later branches.
- This branch does not finish every read-only query surface. Commit graph walking and file-history internals were already moved in earlier stack layers; this PR focuses on the remaining read-only status/metadata/query paths listed below.

What changed:

- Moves workdir/index status collection toward gix:
  - `get_filenames_diff_at_workdir` now opens a gix repo and iterates gix status items.
  - staged changes can be computed from gix head-tree/index comparison.
  - submodule paths are filtered through the new submodule query model.
  - final staged/unstaged/conflict counts are still normalized into the existing `UncommittedChanges` shape.
- Keeps libgit2 where still needed for existing diff/hunk rendering and submodule status compatibility.
- Moves tracked-file search path collection from libgit2 index iteration to gix index iteration.
- Refactors fuzzy file matching to use `SmallVec` and split the position/scoring logic into smaller helpers.
- Moves remote listing/default-remote resolution to gix:
  - remote names and fetch URLs come from gix remote APIs.
  - push URLs and default remote config are read from gix config snapshots.
  - effective default remote still follows the existing priority order.
- Rewrites submodule metadata/status collection around gix:
  - checks for `.gitmodules` in workdir, committed tree, and index.
  - enumerates submodules through gix.
  - records head/index/workdir IDs as `gix::ObjectId`.
  - derives open/uninitialized/config/head/index/workdir state.
  - computes index changes, new commits, modified content, and untracked content.
- Moves worktree metadata listing toward gix:
  - main and linked worktrees are read through gix worktree APIs.
  - worktree heads are stored as `gix::ObjectId`.
  - dirty checking can be skipped for metadata-only callers.
  - current worktree dirty state can be supplied from already-computed uncommitted changes.
- Moves branch visibility’s current branch-name collection to gix reference iteration.
- Updates core model types:
  - reflog alias entry time is now `gix::date::Time`.
  - submodule OIDs are now `gix::ObjectId`.
  - worktree heads are now `gix::ObjectId`.
- Updates query tests for diffs/status, tracked-file search, remotes, submodules, worktrees, branch visibility, and core model behavior.

Algorithm difference:

- Status panes no longer rely on `git2::statuses` as the main source of staged/unstaged file classification.
- Staged-only status can be computed directly from gix head tree plus index.
- Worktree status uses gix status summaries and then maps them back into the existing UI-facing buckets:
  - staged added/modified/deleted
  - unstaged added/modified/deleted
  - conflicts
  - clean/staged/unstaged booleans
- Submodule reporting is now metadata-driven through gix submodule state plus explicit pointer comparisons.
- Remote default selection preserves the existing policy:
  - `guitar.defaultRemote`
  - `remote.pushDefault`
  - current branch upstream remote
  - `origin`
  - first sorted remote
- Worktree listing separates metadata reads from dirty checks, so callers that only need worktree identity can avoid full status scans.
- Branch visibility now reads local and remote branch names through gix refs instead of libgit2 refs.

Performance story:

- Expected wins are mostly from avoiding repeated libgit2 status/config/ref setup in read-only paths.
- Worktree metadata callers can skip dirty checks and reuse already-computed current-worktree dirty state.
- File search avoids libgit2 index iteration and reduces temporary allocation in fuzzy matching.
- Submodule metadata preflight avoids full submodule enumeration when there is no `.gitmodules` metadata.

## Benchmark Results for this PR

Command shape: `hyperfine --warmup 5 './target/release/guitar /home/mjc/src/linux --exit-when-graph-complete'`.

### CPU: M1 MacBook Pro 16GB

| Ref | Total | vs main | User | System | CPU |
| --- | ---: | ---: | ---: | ---: | ---: |
| `04-queries-readonly` | `3.812s` | `41.26x` | `1.47s` | `0.42s` | `49%` |

### CPU: 5950x Linux/ZFS

| Ref | Total | vs main | User | System | CPU |
| --- | ---: | ---: | ---: | ---: | ---: |
| `04-queries-readonly` | `1.856s` | `53.47x` | `1.58s` | `0.46s` | `110%` |

### Memory: 5950x Linux/ZFS

| Ref | Peak heap | Peak RSS |
| --- | ---: | ---: |
| `04-queries-readonly` | `436.12M` | `977.18M` |

<details>
<summary>Top 10 heaptrack allocation sites for `04-queries-readonly`</summary>

1. `190825 calls, 11.40M peak - git__calloc`
2. `101577 calls, 0B peak - <>::next`
3. `100831 calls, 0B peak - gix_dir::walk::readdir::recursive`
4. `89663 calls, 1.27M peak - CRYPTO_malloc`
5. `41425 calls, 1.61M peak - <>::finish_grow`
6. `34119 calls, 0B peak - gix_ref::raw::convert::<>::from`
7. `24684 calls, 0B peak - <>::push_back_tracked_path_component`
8. `17060 calls, 878.18K peak - guitar::core::walker::Walker::new`
9. `13654 calls, 0B peak - std::sys::fs::read_dir`
10. `11856 calls, 154.43M peak - alloc::raw_vec::RawVecInner<>::finish_grow`

</details>

### CODE REVIEW

File: src/app/draw/search.rs [hunk 1]
```diff
@@ -89,7 +89,7 @@ impl App {
                 };
                 lines.push(Line::from(vec![
                     Span::styled(format!("{} ", status_marker(row.status, &self.symbols)), Style::default().fg(marker_color)),
-                    Span::styled(format!("{} ", row.short_oid), Style::default().fg(self.theme.COLOR_GREY_600)),
+                    Span::styled(format!("{:.8} ", row.oid), Style::default().fg(self.theme.COLOR_GREY_600)),
                     Span::styled(truncate_with_ellipsis(&row.summary, summary_width), Style::default().fg(self.theme.COLOR_TEXT)),
```
[the remark this answers is not in the document]
Comment (mjc) [reply src/app/draw/search.rs:93]: bogus comment I am nearly certain

File: src/git/queries/file_history.rs [hunk 2]
```diff
@@ -1,48 +1,244 @@
...
+        .next()
+        .map(|parent_oid| repo.find_commit(parent_oid.detach()).map_err(gix_error).and_then(|parent| parent.tree_id().map(|id| id.detach()).map_err(gix_error)))
+        .transpose()?;
+
+    let ids = (tree_id, parent_tree_id);
+    COMMIT_TREE_CACHE.with(|cache| {
+        cache.borrow_mut().insert(oid, ids);
+    });
+    Ok(ids)
+}
```
[the remark this answers is not in the document]
Comment (mjc) [reply src/git/queries/file_history.rs:107]: worthwhile target for a future pr but I think it was added in an earlier pr in the stack.
